### PR TITLE
feat: make use of updated nim-sqlcipher apis

### LINKED ISSUE #120: Reconcile this repo with changes in nim-sqlcipher
See: https://github.com/status-im/nim-sqlcipher/issues/23.

Involves rebasing and refactoring #113; see also https://github.com/status-im/nim-sqlcipher/issues/20.

### PR DESCRIPTION
**Pull request by michaelsbradleyjr**
Closes #120.

### CODE REVIEW

**Comment by michaelsbradleyjr** [comment]
This PR is now ready for general review; but before it's merged the various WIP commits should be squashed down, so I've added a *don't merge yet* label.

**Review by emizzle — APPROVED** [review]

**Pull request merged**
